app/components/Menu.tsx

```diff
@@ -2,61 +2,105 @@
 
 import Image from 'next/image';
 import iconMenu from '../../assets/image/iconMenu.svg';
-import React, { useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import Link from 'next/link';
 
 export default function Menu() {
   const [openMenu, setOpenMenu] = useState(false);
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  const toglleMenu = () => {
+    setOpenMenu(!openMenu);
+  };
```

Comment:
Keep it one liner (and fix the typo)
```tsx
const toggleMenu = () => setOpenMenu(!openMenu);
```
You'll need to adjust all other places in this file that is wrong too (`toglleMenu` instead of `toggleMenu`)